lua/telescope/pickers.lua

```diff
@@ -336,8 +337,10 @@ function Picker:find()
   local preview_border = preview_opts and preview_opts.border
   self.preview_border = preview_border
 
-  local prompt_win, _, prompt_border_win = self:_create_window("", popup_opts.prompt)
+  local prompt_win, prompt_opts, prompt_border_win = self:_create_window("", popup_opts.prompt)
   local prompt_bufnr = a.nvim_win_get_buf(prompt_win)
+
+  self.prompt_border = prompt_opts and prompt_opts.border
   self.prompt_bufnr = prompt_bufnr
```

Comment:
Might be good to standardise this so things match between results/preview/prompt. The layout you have for results seems like the best way. We can also add `self.preview_win` and `self.prompt_win` at the same time (maybe not for `self.preview_bufnr`, as it changes right?)

Reply:
Agreed, but I suppose in big picture it's more convoluted as it practically makes `telescope.state` a bit vain I suppose. I'd have to grep the code, but my first hunch is we are inconsistent between getting buffers / windows from picker or the (global) state.